### DISCUSSION TITLE
RHF: Watch api, get only the changed field

### DISCUSSION THREAD

**Discussion by senal** [react-hook-form/react-hook-form#5755] [Q&A]
Hi,
For an instance, I have a form of 5 fields, I just need a way of select/retrieves only the field that has been changed by the user.
Usually, this can be achieved through the onChange(..) event. 

But how do I achieve this using the 'watch' API?

// this is how I register my default values using useForm 
 const methods = useForm({
        defaultValues: {
            firstName: 'Kim',
            lastName: 'Lowe'
            entityName: 'Me and You',
            tradingName: 'Me and You',
            comments: "Bla"
        }
    });

// components are wrapped with FormProvider
<FormProvider {...methods}>
            <ChildOne></ChildOne>
            <ChildTwo></ChildTwo>
            <button onClick={ () =>  methods.handleSubmit(onSave)() }>Save</button>
     </FormProvider>

I tried below:

methods.watch((data, {name, type}) => {
        console.log(name, type);
});

When I change one field value, I see so many console logs of the changed filed name. why do I see many logs and not why single field value change?

Please note that I don't want to pass the list of parameters to the 'watch' API.

Any suggestions?

Thanks

**Comment by bluebill1049** [comment]
at the moment, by default we return the entire form value.

**Reply by senal** [reply]
Thanks @bluebill1049, for your prompt feedback.
Would this be a potential candidate for a new feature?
